scalafix-cli/src/main/scala/scalafix/internal/v1/MainOps.scala

```diff
@@ -347,6 +348,18 @@ object MainOps {
     } else {
       args.rules.syntacticPatch(doc, args.args.autoSuppressLinterErrors)
     }
+    // close global if evaluated
+    cleanGlobalRessources(args.global)
+    results
+  }
+
+  private def cleanGlobalRessources(global: LazyValue[Option[Global]]): Unit = {
+    if (global.isEvaluated)
+      global.value.foreach { g =>
+        g.askShutdown()
+        g.closeCompat()
+      }
```

Comment:
is that the prefered way to shut down the compiler? https://github.com/scalacenter/scalafix/blob/f6bff33bf13512a9e73c83e448eb5456de1b5db2/scalafix-rules/src/main/scala/scalafix/internal/rule/ExplicitResultTypes.scala#L38-L47 is a bit different

Reply:
ok

Reply:
those two methods are not supposed to throw exceptions. but why not

Reply:
I didn't look at the git history, but I am guessing this was done for a reason? or to be defensive, which is not a bad thing